Eggstravaganza/Assets/Scripts/ServerStartup.cs

```diff
@@ -0,0 +1,251 @@
+// Inspired by samyam's video on the topic: https://www.youtube.com/watch?v=FjOZrSPL_-Y&t=1052s
+
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Unity.Netcode;
+using Unity.Netcode.Transports.UTP;
+using Unity.Services.Core;
+using Unity.Services.Matchmaker;
+using Unity.Services.Matchmaker.Models;
+using Unity.Services.Multiplay;
+using UnityEngine;
+
+public class ServerStartup : MonoBehaviour
+{
+    public static event Action ClientInstance;
+    
+    private const int k_MaxPlayers = 4;
+    private const int k_MultiplayServiceTimeout = 20000;
+    private const string k_InternalServerIp = "0.0.0.0";
+    private const int k_TicketCheckMs = 1000;
+
+    private ushort m_ServerPort = 7777;
+    private string m_ExternalServerIp = "0.0.0.0";
+    private string m_ExternalConnectionString => $"{m_ExternalServerIp}:{m_ServerPort}";
+
+    private IMultiplayService m_MultiplayService;
+    private string m_AllocationId;
+    private MultiplayEventCallbacks m_ServerCallbacks;
+
+    private BackfillTicket m_LocalBackfillTicket;
+    private CreateBackfillTicketOptions m_CreateBackfillTicketOptions;
+    private MatchmakingResults m_MatchmakingPayload;
+    private bool m_Backfilling;
+
+    private async void Start()
+    {
+        var server = false;
+        var args = Environment.GetCommandLineArgs();
+        for (var i = 0; i < args.Length; i++)
+        {
+            if (args[i] == "-dedicatedServer")
+            {
+                server = true;
+            }
```

Comment:
just for my understanding, so when you're launching on multiplay, you are literally running a linux build with this flag in some field in the multiplay settings to start a server?

Reply:
Yep yep exactly, it's in the build config! I also tell it to use batchmode and no rendering there